client/network/abortable-thunk.ts

```diff
@@ -17,12 +17,12 @@ export interface RequestHandlingSpec<T = void> {
   /**
    * A function that will be called if the underlying request succeeds.
    */
-  onSuccess: (result: T) => void
+  onSuccess?: (result: T) => void
```

Comment:
I didn't make these optional because it makes it very easy to use this API wrong, would prefer you don't change that